test/bundler/expectBundled.ts

```diff
@@ -1247,6 +1247,27 @@ for (const [key, blob] of build.outputs) {
       }
     }
 
+    // Check that all source maps are valid JSON
+    if (opts.sourceMap === "external" && outdir) {
+      for (const file of readdirSync(outdir)) {
+        if (file.endsWith(".map")) {
+          const parsed = JSON.parse(readFileSync(path.join(outdir, file)).toString());
+          try {
+            expect(parsed).toBeDefined();
+            expect(parsed.mappings).toBeString();
+            expect(parsed.sources).toBeArray();
+            expect(parsed.sourcesContent).toBeArray();
```

Comment:
want to make sure these are non-empty arrays no?

Reply:
they technically are allowed to be empty if the file doesnt point to source code (such as generated helper functions like `__commonJS` or `__using`).

now that you mention it, i honestly should throw this through an actual sourcemap parser to ensure all the remappings parse fine too.